File: bomberman.ui/src/main/app/game/game.service.js
export default class GameService {

    constructor($http, $log) {
        "ngInject";
        this.http = $http;
        this.log = $log;
    }

    createNewGame() {
        return this.http.post("/bomberman/api/games", {});
    }

    connectToGame(gameId) {
        return this.http.put("/bomberman/api/games/" + gameId, {});
    }

    sendCommand(gameId, playerId, command) {
        const url = "/bomberman/api/game/" + gameId + "/player/" + playerId + "/command";
        return this.http.post(url, {
            command: command
        });
    }
}

File: bomberman.server/src/main/java/com/uawebchallenge/bomberman/game/control/GameService.java
package com.uawebchallenge.bomberman.game.control;

import com.uawebchallenge.bomberman.game.exception.BombermanException;
import com.uawebchallenge.bomberman.game.model.player.PlayerCommand;

public interface GameService {

    NewGameData createNewGame() throws BombermanException;

    void addCommand(String gameId, String playerId, PlayerCommand playerCommand) throws BombermanException;

    NewGameData connectToGame(String gameId) throws BombermanException;
}
